.github/workflows/ci_cd.yml

```diff
@@ -33,6 +33,9 @@ jobs:
       - name: Install Python tools
         run: pip install build twine
 
+      - name: Install QPAX
+        run: pip install git+https://github.com/kevin-tracy/qpax
+
```

Comment:
As soon as we are sure enough of this PR also considering the performance of the new contact model and this QP library, we should start considering packaging `qpax` at least in conda-forge. I'd keep for the moment the imports private (i.e. not in the top level but inside function's bodies) so that we can mark it as optional dependency and not being blocked.

Reply:
I had a look and it seems that an external contributor already proposed in https://github.com/kevin-tracy/qpax/pull/2 a CD workflow towards PyPI, but at the time of writing there is no uploaded package. Let's keep it under our radar.

Reply:
Thanks! Ok perfect! I could try to package it in conda forge in the next weeks :) 

For the time being, how do you suggest to proceed? I will make the import  private in the source code, but I don't know where to install the package in order to have the CICD working (it seems that those lines I added didn't worked out).

Reply:
If the imports are private and there is no test against this new functionality, CI should work without any modification. It's a good starting point to check that the refactoring did not break anything.

As soon as you add a test, you need `qpax` in github actions. You added the lines for installing it in the wrong job. You should add it [here](https://github.com/ami-iit/jaxsim/blob/1145d1f91f19726e904e313b3ad642288a1ed90c/.github/workflows/ci_cd.yml#L92) (in the `test` job, not the `package` one).

> Ok perfect! I could try to package it in conda forge in the next weeks :)

I'd wait a little longer to check if a new release will be pushed to PyPI soon. It's easier packaging in conda-forge if it is already distributed in PyPI.

Reply:
> Thanks! Ok perfect! I could try to package it in conda forge in the next weeks :)

Great, feel free to let me know if you need help.